src/sources/util/tcp.rs

```diff
@@ -15,6 +15,7 @@ use tokio::{
     reactor::Handle,
     timer,
 };
+use tokio_tls::TlsAcceptor;
```

Comment:
do we want to just start using `tokio_openssl` instead?

Reply:
That may make sense. However, given that nothing else in Vector uses `tokio_openssl` yet, but other modules use `tokio_tls` as well (ex the parallel `src/sinks/utils/tcp.rs'), I think it would make more sense as a separate PR.

Reply:
Well that isn't true, I did add this :P https://github.com/timberio/vector/blob/ce326530da9235233355a86dcc5c27b626f40a10/src/sinks/util/rusoto.rs#L7

We really need to just pick one I think if we are already here we should just do it now instead of punting imo.

Reply:
Uh, ok, you said `tokio_openssl`, so I grepped the sources for `tokio_openssl` direct uses, of which there are still none.

I can rework the sources here, but are you also advising me to rework the sinks as well to use `tokio_openssl` in this PR?

Reply:
No just the stuff you touch in this PR. And sorry I totally could have been more clear. `hyper-openssl` and `tokio-openssl` are basically just wrappers around each other that implement the correct traits for the `openssl` crate. So I think it makes sense to write any new code using that then going back later and updating the others.

Reply:
Ok, so I looked into this, and it requires a rework of the TLS config handling that will end up being easier if I just do both acceptors and connectors at once, due to the different types involved (`native_tls` wrappers vs direct use of `openssl`).

Reply:
That works for me :+1: